runway/cfngin/hooks/aws_lambda.py

```diff
@@ -163,6 +173,176 @@ def _zip_from_file_patterns(root, includes, excludes, follow_symlinks):
+                                          stderr=subprocess.PIPE)
+        _stdout, stderr = pipenv_process.communicate(timeout=timeout)
+        if pipenv_process.returncode == 0:
+            return req_path
+        if int(sys.version[0]) > 2:
+            stderr = stderr.decode('UTF-8')
+        LOGGER.error('"%s" failed with the following output:\n%s',
+                     ' '.join(cmd), stderr)
+        raise PipenvError
+
+
+def dockerized_pip(work_dir, runtime=None, docker_file=None,
+                   docker_image=None, **_kwargs):
+    """Run pip with docker.
+
+    Args:
+        work_dir (str): Work directory for docker.
+        runtime (Optional[str]): Lambda runtime. Must provide one of
+            ``runtime``, ``docker_file``, or ``docker_image``.
+        docker_file (Optional[str]): Path to a Dockerfile to build an image.
+            Must provide one of ``runtime``, ``docker_file``, or
+            ``docker_image``.
+        docker_image (Optional[str]): Local or remote docker image to use.
+            Must provide one of ``runtime``, ``docker_file``, or
+            ``docker_image``.
+        kwargs (Any): Advanced options for docker. See source code to
+            determine what is supported.
+
+    Returns:
+        Tuple[str, str]: Content of the ZIP file as a byte string and
+        calculated hash of all the files
+
+    """
+    # TODO use kwargs to pass args to docker for advanced config
+    # TODO stream docker logs to stdout/stderr
+    if docker_file and docker_file and runtime:
+        raise InvalidDockerizePipConfiguration(
+            'only one of [docker_file, docker_file, runtime] can be specified'
+        )
+    docker = Docker.from_env()
+
+    if docker_file:
+        if not os.path.isfile(docker_file):
+            raise ValueError('could not find docker_file "%s"' % docker_file)
+        tmp_image, _build_logs = docker.images.build(
+            path=os.path.dirname(docker_file),
+            dockerfile=os.path.basename(docker_file)
+        )
+        docker_image = tmp_image.id
+
+    if runtime:
+        docker_image = 'lambci/lambda:build-%s' % runtime
+
+    if not docker_image:
+        raise InvalidDockerizePipConfiguration(
+            'one of [docker_file, docker_file, runtime] must be specified'
+        )
+
+    if sys.platform.lower() == 'win32':
+        work_dir = work_dir.replace('\\', '/')
+
+    work_dir_mount = Docker.types.Mount(target='/var/task',
+                                        source=work_dir,
+                                        type='bind')
+    pip_cmd = (
+        'python -m pip install -t /var/task -r /var/task/requirements.txt'
+    )
+
+    # TODO figure out what to do with the logs returned here or drop the var
+    _ = docker.containers.run(image=docker_image,
+                              command=['/bin/sh', '-c', pip_cmd],
+                              auto_remove=True,
+                              mounts=[work_dir_mount])
```

Comment:
i think this section can be replaced with this to stream the logs. `stream=True` will return a generator we can iterate over to log the response.

```suggestion
    logs = docker.containers.run(image=docker_image,
                                 command=['/bin/sh', '-c', pip_cmd],
                                 auto_remove=True,
                                 mounts=[work_dir_mount],
                                 stream=True)

    for line in logs:
        LOGGER.info(line)
```